pkg/kubelet/deviceplugin/manager.go

```diff
@@ -212,14 +212,8 @@ func (m *ManagerImpl) addEndpoint(r *pluginapi.RegisterRequest) {
 		return
 	}
 
-	stream, err := e.list()
-	if err != nil {
-		glog.Errorf("Failed to List devices for plugin %v: %v", r.ResourceName, err)
-		return
-	}
-
 	go func() {
-		e.listAndWatch(stream)
+		e.run()
```

Comment:
This will re-open issue #52510

Reply:
So I was thinking the following might be a solution:

```golang
// run guarantees that if error is not nil, it has received at least one call to ListAndWatch by the client Device Plugin 
func (e *endpoint) run() error {
	callback := e.callback
	deviceChan := make(chan interface{})

	e.callback = func(n string, a, u, r []*pluginapi.Device) {
		callback(n, a, u, r)
		e.callback = callback

		close(deviceChan)
	}
	
	go e.listAndWatch()
	<-deviceChan

     return nil
}
```

Reply:
Well, I'm not sure about your meaning. Is it in order to prevent that the plugin newly registered could not listAndWatch successfully at all, so that the the old devices in handler & node capacity could not be deleted correspondingly?
I don't think the change you suggested could take effect on it. How would it behavior if an error occurs? Lead to a block?
How about move `e.callback(e.resourceName, []*pluginapi.Device{}, []*pluginapi.Device{}, e.getDevices())` into `e.stop()`? since the stop func is thoutht to be a cleanup func of endpoint